src/Bicep.Core/Syntax/SyntaxHelper.cs

```diff
@@ -34,12 +35,19 @@ public static class SyntaxHelper
             IArtifactReferenceSyntax foreignTemplateReference,
             DiagnosticBuilder.ErrorBuilderDelegate onUnspecifiedPath)
         {
-            if (foreignTemplateReference.Path is not StringSyntax pathSyntax)
+            if (foreignTemplateReference.Path is not StringSyntax && foreignTemplateReference.Path is not NoneLiteralSyntax)
             {
                 return new(onUnspecifiedPath);
             }
 
-            if (pathSyntax.TryGetLiteralValue() is not string pathValue)
+            if (foreignTemplateReference.Path is NoneLiteralSyntax)
+            {
+                return new(string.Empty);
```

Comment:
Returning `""` feels like you're trying to avoid a larger refactor by "making the compiler happy". This is a form of subtle tech debt, because a caller can no longer rely on this method to return a valid path.

I would want to see if there's a way we can avoid this - possibly by handling the check for `none` at a higher level in the code.

Reply:
with the latest code changes, this piece became unnecessary, and removed 👍